internal/seclang/directives.go

```diff
@@ -952,22 +952,101 @@ func directiveSecDebugLogLevel(options *DirectiveOptions) error {
 	return options.WAF.SetDebugLogLevel(debuglog.Level(lvl))
 }
 
+// Description: Updates the target (variable) list of the specified rule(s).
+// Syntax: SecRuleUpdateTargetById ID TARGET1[|TARGET2|TARGET3]
+// ---
+// This directive will append variables to the specified rule with the targets provided in the second parameter.
+// The rule ID can be single IDs or ranges of IDs. The targets are separated by a pipe character.
 func directiveSecRuleUpdateTargetByID(options *DirectiveOptions) error {
-	idStr, v, ok := strings.Cut(options.Opts, " ")
-	if !ok {
+	if len(options.Opts) == 0 {
+		return errEmptyOptions
+	}
+
+	idsOrRanges := strings.Fields(options.Opts)
+	length := len(idsOrRanges)
+	if length < 2 {
 		return errors.New("syntax error: SecRuleUpdateTargetById id \"VARIABLES\"")
 	}
-	id, err := strconv.Atoi(idStr)
-	if err != nil {
-		return err
+	// The last element is expected to be the variable(s)
+	variables := idsOrRanges[length-1]
+	for _, idOrRange := range idsOrRanges[:length-1] {
+		if idx := strings.Index(idOrRange, "-"); idx == -1 {
+			id, err := strconv.Atoi(idOrRange)
+			if err != nil {
+				return err
+			}
+
+			rule := options.WAF.Rules.FindByID(id)
+			if rule == nil {
+				return fmt.Errorf("SecRuleUpdateTargetById: rule \"%d\" not found", id)
+			}
+			rp := RuleParser{
+				rule:           rule,
+				options:        RuleOptions{},
+				defaultActions: map[types.RulePhase][]ruleAction{},
+			}
+			return rp.ParseVariables(strings.Trim(variables, "\""))
+
+		} else {
+			start, err := strconv.Atoi(idOrRange[:idx])
```

Comment:
What if start < 0?

Reply:
It was implicitly already handled because `idx` ends up being `0` and failing Atoi. Still, I added an explicit check to provide a better error message